lib/websocket_server.rb

```diff
@@ -1,7 +1,11 @@
 class WebsocketServer
+  attr_accessor :logger
+
   Pairing = Struct.new(:is_ws, :proxy)
 
-  def initialize
+  def initialize(options = {})
+    @logger = options.fetch(:logger, $websocket_log)
+    logger.info('Initializing websocket worker!')
```

Comment:
@skateman This initialize is getting really complicated, maybe for future PRs, you can think about extracting a method for starting the socket reading thread and also ensure that thread gets shutdown properly as part of a shutdown method.  Maybe?  Also, do we expect to only create one instance of this class per process?  Should this be a singleton?   Just ideas for the future...